Parse/src/test/java/com/parse/ParseKeyValueCacheTest.java

```diff
@@ -59,4 +65,33 @@ public Void call() throws Exception {
     }
     ParseTaskUtils.wait(Task.whenAll(tasks));
   }
+
+  @Test
+  public void testSaveToKeyValueCacheWithoutCacheDir() throws Exception {
+    // Delete the cache folder(Simulate users clear the app cache)
+    File keyValueCacheDir = new File(temporaryFolder.getRoot(), "ParseKeyValueCache");
```

Comment:
nit: It would be better to hold a reference to the `File` we use in `initialize()` in `setUp` and use that here.